bin/git-forgit

```diff
@@ -677,9 +709,15 @@ _forgit_branch_delete() {
     "
 
     cmd="git branch --color=always | LC_ALL=C sort -k1.1,1.1 -rs"
```

Comment:
What about this deferred `git` call here? Can we get rid of that as well?

Reply:
I've only included deferred `git` calls that we replaced with separate functions in #326, see my [previous comment](https://github.com/wfxr/forgit/pull/343#issuecomment-1951397215). We could replace all deferred `git` calls in this PR, which would make it a bit bigger, as there are still a few of them left that we did not create functions for. We could also do that in the next PR. I'm open for both.

Reply:
Ah sorry, I missed that part of the comment in the first place. I would prefer replacing all `git` calls in this PR, since that's what its title says. :-)

Reply:
f79da92 removes all deferred git commands. I had to include converting `$_forgit_emojify` into a function and the replacement of the `$graph` variable with an array to get the commands working without deferred code and `eval`. This makes this PR quite big. Haven't squashed the commit yet, because I'm thinking creating a separate PR out of f79da92 to reduce the size of this one. WDYT?

Reply:
Ah, I see, that's quite a huge additional change indeed. I did not expect that, sorry. So yes, let's move that to a new PR. I would consider this PR finished then.

Do I get it right that none of the new changes were part of the original #326? I wonder what's the best way to continue then, since we have some changes in #326 left to include. Shall we do those first?

Reply:
> Do I get it right that none of the new changes were part of the original https://github.com/wfxr/forgit/pull/326?

No they were mostly part of #326, I just moved them over to this branch. The only thing in this PR that is not in #326 yet are the four hunks that remove `xargs`. 
I would suggest we rebase #326 on this PR to make sure we don't digress too far and I create a new PR that converts `$_forgit_emojify` into a function and after that a PR that replaces the rest of the deferred `git` commands. Sounds good?

Reply:
Yep, that sounds good, thanks a lot!

Reply:
FYI: I started to rebase #326. I'm almost done and will finalize it tomorrow, so don't put any effort into it.

Reply:
Oh wow, thanks a lot for your effort!